fdbserver/workloads/BackupCorrectnessPartitioned.actor.cpp

```diff
@@ -0,0 +1,797 @@
+			BackupAndRestorePartitionedCorrectnessWorkload::backupAgentRequests++;
+		}
+
+		if (self->encryptionKeyFileName.present()) {
+			wait(BackupContainerFileSystem::createTestEncryptionKeyFile(self->encryptionKeyFileName.get()));
+		}
+
+		try {
+			state Future<Void> startRestore = delay(self->restoreAfter);
+
+			// backup
+			wait(delay(self->backupAfter));
+
+			TraceEvent("BARW_DoBackup1", randomID).detail("Tag", printable(self->backupTag));
+			state Future<Void> b =
+			    doBackup(self, 0, &backupAgent, cx, self->backupTag, self->backupRanges, self->stopDifferentialAfter);
+
+			TraceEvent("BARW_DoBackupWait", randomID)
+			    .detail("BackupTag", printable(self->backupTag))
+			    .detail("AbortAndRestartAfter", self->abortAndRestartAfter);
+			try {
+				wait(b);
+			} catch (Error& e) {
+				if (e.code() != error_code_database_locked)
+					throw;
+				if (self->performRestore)
+					throw;
+				return Void();
+			}
+			TraceEvent("BARW_DoBackupDone", randomID)
+			    .detail("BackupTag", printable(self->backupTag))
+			    .detail("AbortAndRestartAfter", self->abortAndRestartAfter);
+
+			state KeyBackedTag keyBackedTag = makeBackupTag(self->backupTag.toString());
+			UidAndAbortedFlagT uidFlag = wait(keyBackedTag.getOrThrow(cx.getReference()));
+			state UID logUid = uidFlag.first;
+			state Key destUidValue = wait(BackupConfig(logUid).destUidValue().getD(cx.getReference()));
+			state Reference<IBackupContainer> lastBackupContainer =
+			    wait(BackupConfig(logUid).backupContainer().getD(cx.getReference()));
+
+			CODE_PROBE(!startRestore.isReady(), "Restore starts at specified time");
+			wait(startRestore);
+			// between here
+			if (lastBackupContainer && self->performRestore) {
+				auto container = IBackupContainer::openContainer(lastBackupContainer->getURL(),
+				                                                 lastBackupContainer->getProxy(),
+				                                                 lastBackupContainer->getEncryptionKeyFileName());
+				BackupDescription desc = wait(container->describeBackup());
+				state Version targetVersion = -1;
+				if (desc.maxRestorableVersion.present()) {
+					if (deterministicRandom()->random01() < 0.1) {
+						targetVersion = desc.minRestorableVersion.get();
+					} else if (deterministicRandom()->random01() < 0.1) {
+						targetVersion = desc.maxRestorableVersion.get();
+					} else if (deterministicRandom()->random01() < 0.5) {
+						targetVersion = deterministicRandom()->randomInt64(desc.minRestorableVersion.get(),
+						                                                   desc.contiguousLogEnd.get());
+					}
+				}
+				wait(runRYWTransaction(cx, [=](Reference<ReadYourWritesTransaction> tr) -> Future<Void> {
+					tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+					for (auto& kvrange : self->backupRanges) {
+						// version needs to be decided before this transaction otherwise
+						// this clear mutation might be backup as well
```

Comment:
Can you fix the rare bug of this test? I.e., the `targetVersion` must be less than this transaction's commit version? This can be done in a separate PR.

Reply:
this is already fixed, targetVersion is selected before this transaction.